app/src/main/java/com/example/next_show/adapters/ShowAdapter.java

```diff
@@ -55,6 +60,22 @@ public void clear() {
 
     public void addAll(List<Show> list) {
         shows.addAll(list);
+        allShows = shows;
+        notifyDataSetChanged();
+    }
+
+    public void filter(String query){
+        Log.i("Adapter", "Filtering for " + query);
+        List<Show> queriedShows = new ArrayList<>();
+        for(Show s : shows){
+            if (s.getGenres().contains(query)) {
+                queriedShows.add(s);
+            }
+        }
+
+        // update the adapter and clear current items
+        shows.clear();
+        shows.addAll(queriedShows);
         notifyDataSetChanged();
```

Comment:
Might be cleaner if your adapter isn't aware of filtering and just told a new list of shows -- up to you, though